crates/bevy_core_pipeline/src/deferred/node.rs

```diff
@@ -0,0 +1,205 @@
+                    resolve_target: None,
+                    ops: Operations {
+                        #[cfg(all(feature = "webgl", target_arch = "wasm32"))]
+                        load: LoadOp::Load,
+                        #[cfg(not(all(feature = "webgl", target_arch = "wasm32")))]
+                        load: LoadOp::Clear(Default::default()),
+                        store: true,
+                    },
+                }),
+        );
+
+        color_attachments.push(
+            view_prepass_textures
+                .deferred_lighting_pass_id
+                .as_ref()
+                .map(|deferred_lighting_pass_id| RenderPassColorAttachment {
+                    view: &deferred_lighting_pass_id.default_view,
+                    resolve_target: None,
+                    ops: Operations {
+                        load: LoadOp::Clear(Default::default()),
+                        store: true,
+                    },
+                }),
+        );
+
+        if color_attachments.iter().all(Option::is_none) {
+            // All attachments are none: clear the attachment list so that no fragment shader is required.
+            color_attachments.clear();
+        }
+
+        {
+            // Set up the pass descriptor with the depth attachment and optional color attachments.
+            let mut render_pass = render_context.begin_tracked_render_pass(RenderPassDescriptor {
+                label: Some("deferred"),
+                color_attachments: &color_attachments,
+                depth_stencil_attachment: Some(RenderPassDepthStencilAttachment {
+                    view: &view_depth_texture.view,
+                    depth_ops: Some(Operations {
+                        load: if depth_prepass.is_some()
+                            || normal_prepass.is_some()
+                            || motion_vector_prepass.is_some()
+                        {
+                            // If any prepass runs, it will generate a depth buffer so we should use it.
+                            Camera3dDepthLoadOp::Load
+                        } else {
+                            // NOTE: 0.0 is the far plane due to bevy's use of reverse-z projections.
+                            camera_3d.depth_load_op.clone()
+                        }
+                        .into(),
+                        store: true,
+                    }),
+                    stencil_ops: None,
+                }),
+            });
+
+            if let Some(viewport) = camera.viewport.as_ref() {
+                render_pass.set_camera_viewport(viewport);
+            }
+
+            // Always run opaque pass to ensure screen is cleared.
+            {
+                // Run the prepass, sorted front-to-back.
+                #[cfg(feature = "trace")]
+                let _opaque_prepass_span = info_span!("opaque_deferred").entered();
+                opaque_deferred_phase.render(&mut render_pass, world, view_entity);
+            }
```

Comment:
Depending on whether deferred or forward is run first, the first node needs to run its first pass always in order to clear, the rest don't need to.

Reply:
This is actually making sure the gbuffer and deferred_lighting_pass_id are cleared. Not the color target. Updated comment.